consensus-types/helpers/comparisons.go

```diff
@@ -0,0 +1,109 @@
+package helpers
+
+import (
+	"bytes"
+
+	ethpb "github.com/OffchainLabs/prysm/v6/proto/prysm/v1alpha1"
+)
+
+func ForksEqual(s, t *ethpb.Fork) bool {
+	if s == nil && t == nil {
+		return true
+	}
+	if s == nil || t == nil {
+		return false
+	}
+	if s.Epoch != t.Epoch {
+		return false
+	}
+	if !bytes.Equal(s.PreviousVersion, t.PreviousVersion) {
+		return false
+	}
+	return bytes.Equal(s.CurrentVersion, t.CurrentVersion)
+}
+
+func BlockHeadersEqual(s, t *ethpb.BeaconBlockHeader) bool {
+	if s == nil && t == nil {
+		return true
+	}
+	if s == nil || t == nil {
+		return false
+	}
+	if s.Slot != t.Slot {
+		return false
+	}
+	if s.ProposerIndex != t.ProposerIndex {
+		return false
+	}
+	if !bytes.Equal(s.ParentRoot, t.ParentRoot) {
+		return false
+	}
+	if !bytes.Equal(s.StateRoot, t.StateRoot) {
+		return false
+	}
+	return bytes.Equal(s.BodyRoot, t.BodyRoot)
+}
+
+func Eth1DataEqual(s, t *ethpb.Eth1Data) bool {
+	if s == nil && t == nil {
```

Comment:
This repeated across all the functions could be a generic I think why do i also feel like we should already ahve this kind of function or maybe golang provids something like this